src/wp-includes/class-wp-block.php

```diff
@@ -498,6 +462,54 @@ private function replace_html( string $block_content, string $attribute_name, $s
 		}
 	}
 
+	private static function get_block_bindings_processor( string $block_content ) {
+		static $internal_processor_class = null;
+		if ( null === $internal_processor_class ) {
+			$internal_processor_class = new class('', WP_HTML_Processor::CONSTRUCTOR_UNLOCK_CODE) extends WP_HTML_Processor {
+				/**
+				 * Replace the rich text content between a tag opener and matching closer.
+				 *
+				 * When stopped on a tag opener, replace the content enclosed by it and its
+				 * matching closer with the provided rich text.
+				 *
+				 * @param string $rich_text The rich text to replace the original content with.
+				 * @return bool True on success.
+				 */
+				public function replace_rich_text( $rich_text ) {
+					if ( $this->is_tag_closer() ) {
+						return false;
+					}
+
+					$depth = $this->get_current_depth();
+
+					$this->set_bookmark( '_wp_block_bindings_tag_opener' );
+					// The bookmark names are prefixed with `_` so the key below has an extra `_`.
+					$bm    = $this->bookmarks['__wp_block_bindings_tag_opener'];
```

Comment:
Should this have a more descriptive name? `$block_bindings_bookmark_opener`

Reply:
https://github.com/WordPress/wordpress-develop/pull/9469/commits/4dbb0e5024f32f2ed7cc3774e6c992736cd60f9d